components/tikv_util/src/file.rs

```diff
@@ -109,7 +109,7 @@ impl<R> Sha256Reader<R> {
 
 impl<R: Read> Read for Sha256Reader<R> {
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        let len = self.reader.read(buf)?;
+        let len = self.reader.lock().unwrap().read(buf)?;
```

Comment:
Why need to lock it? Isn't `self` is borrowed exclusively?

Reply:
`ExternalStorage::write` now require `Sync` trait, which is because rusoto's `PutObjectRequest::body` require `Sync` trait. The `reader` here is a `DBSequentialFile`, which is not thread-safe.

Reply:
are you suggesting `unsafe impl Sync for Sha256Reader`?

Reply:
I suggest to require `R` to be `Sync`, so that if reader is Sync already, it doesn't have to pay for the cost of mutex. By the way, why is `DBSequentialFile` not `Sync`?

Reply:
rocksdb `SequentialFile` maintains some internal state and is not thread-safe, so marking it `Sync` is improper, from my understanding.

Reply:
I changed to `get_mut` instead of `lock` to avoid actual locking, are you okay with it?